MantidPlot/src/ProjectRecovery.h

```diff
@@ -114,6 +115,10 @@ class ProjectRecovery {
   // Return true if the folder at the end of the path is older than a month.
   bool olderThanAGivenTime(const Poco::Path &path, int64_t elapsedTime);
 
+  // Remove Empty WorkspaceGroups from the ADS and passed vector
+  void removeEmptyGroupsFromADS(
```

Comment:
This does not act on internal state. Should not be a member function.

Reply:
I'm not sure why that is a necessary requirement, after all it's a private member function so it's basically the same thing as a function declared in an anonymous namespace and in this case I get to define it closer to the one function that uses it?

Reply:
Closer means that I put the actual code directly below it so it is much easier to find and not at the top of the file in the anonymous namespace. I tried to find some links and I think it would be helpful to me for better understanding. Also with regards to your point 1, it doesn't change what is actually exposed outside of my class/type as it's a private member function, I believe that's where we disagree.

Reply:
In a modern IDE, I don't think your code proximity has any relevance for navigation. Yes, you have to declare the function before you use it, but the declaration can be declared anywhere near your current member definition. So if proximity is an issue, you can put it directly above.

I think you're confusing the logical and physical parts of the c++ system. logically, yes this is private so it cannot be invoked from outside the object to which it belongs. However physically it gets seen as part of the pre-processor expansion of the include. i.e. any translation unit including this header will have to be recompiled. This is the whole reasoning behind PIMPL

Effective c++ item 23 (should be copies in the main office) by Scott Meyers offer a good explanation as to the topic of preferring non-member functions.